### PR TITLE
Add Command Base

### PR DESCRIPTION
**Pull request by rypete**
Paves way to add commands from the server.  A sample set of commands has been added:

`!sr set team [team name]` Sets team name.  Admin only
`!sr team`: Returns team name

`set` commands are available to server admins only.  Server `admins: string[]` is a new property on the server objects in the config file.  Place the username of admins in it.

In other news, the console is now colored, and I've added vsc debugging support

### CODE REVIEW

**Comment by rypete** [comment]
Things are getting bigger now - it would be good to validate architecture.  I've been trying to keep commands/comms in `SrBot` and file rw and data in `statsgenerator`

File: src/SrBot.ts [hunk 1]
```diff
@@ -48,57 +77,143 @@ export class SrBot {
...
+                switch (params[0].toUpperCase()) {
+                    case UPDATABLE_COMMAND.TEAM: {
+                        // !sr set team ####
+                        const teamName = settableParams.join(' ');
+                        this.statsGenerator.updateServerProperty(serverId, 'teamName', teamName);
+                        this.sendToChannel(message.channel, `Team name updated: ${teamName}`);
+                        break;
+                    }
+                    case UPDATABLE_COMMAND.REGION: {
+                        // !sr set region ####
```
Comment (Derekholio) [inline src/SrBot.ts:180]: Validate selected region here us|eu|kr|cn|global

File: src/SrBot.ts [hunk 2]
```diff
@@ -48,57 +77,143 @@ export class SrBot {
...
+                        this.statsGenerator.updateServerProperty(serverId, 'region', settableParams[0].toLowerCase() as OverwatchAPI.REGION);
+                        break;
+                    }
+                    case UPDATABLE_COMMAND.PLATFORM: {
+                        // !sr set platform ####
```
Comment (Derekholio) [inline src/SrBot.ts:192]: Validate selected platform here pc|xbl|psn

**Review by Derekholio — CHANGES_REQUESTED** [review]

File: src/SrBot.ts (hunk 1 above)
Comment (rypete) [reply src/SrBot.ts:180]: Now validated

File: src/SrBot.ts (hunk 2 above)
Comment (rypete) [reply src/SrBot.ts:192]: Now validated

**Review by Derekholio — APPROVED** [review]

**Comment by Derekholio** [comment]
Completed #17 #18 #19

**Pull request merged**